### PR TITLE
Introduce reset subcommand for widget

### LINKED ISSUE #3063: Introduce reset subcommand for sidebar
Proposal:

`wp sidebar reset <sidebar-id>`

- Cleans up all widgets in the the given sidebar.
- If sidebar ID is not passed, all registered sidebars will be reset.

Technically, reset means moving widgets from sidebar to `Inactive Widgets`.

### PR DESCRIPTION
**Pull request by ernilambar**
Initially at https://github.com/wp-cli/wp-cli/issues/3063, reset was supposed to be kept under sidebar. But after https://github.com/wp-cli/wp-cli/pull/3071 reset feature is kept under widget command. 

Fixes #3063

### CODE REVIEW

File: php/commands/widget.php [hunk 1]
```diff
@@ -305,6 +305,81 @@ public function delete( $args, $assoc_args ) {
 	}
 
 	/**
+	 * Reset sidebar.
+	 *
+	 * Removes all widgets from the sidebar and place those to Inactive Widgets.
```
Comment (danielbachhuber) [inline php/commands/widget.php]: Change "place those to" to "places them in"

File: features/widget-reset.feature [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    And I run `wp widget list sidebar-2 --format=count`
+    Then STDOUT should be:
+      """
+      0
+      """
+    And I run `wp widget list sidebar-3 --format=count`
+    Then STDOUT should be:
+      """
+      0
+      """
```
Comment (danielbachhuber) [inline features/widget-reset.feature]: Can we verify the widgets were added to the inactive sidebar?

File: php/commands/widget.php [hunk 3]
```diff
@@ -305,6 +305,81 @@ public function delete( $args, $assoc_args ) {
...
+
+		foreach ( $args as $sidebar_id ) {
+			if ( ! array_key_exists( $sidebar_id, $wp_registered_sidebars ) ) {
+				WP_CLI::warning( sprintf( 'Invalid sidebar: %s', $sidebar_id ) );
+				continue;
+			}
+
+			$widgets = $this->get_sidebar_widgets( $sidebar_id );
+			if ( empty( $widgets ) ) {
+				WP_CLI::warning( sprintf( "'%s' is already empty.", $sidebar_id ) );
```
Comment (danielbachhuber) [inline php/commands/widget.php]:
```
Sidebar '%s' is already empty.
```

File: php/commands/widget.php [hunk 4]
```diff
@@ -305,6 +305,81 @@ public function delete( $args, $assoc_args ) {
...
+			$args = array_keys( $wp_registered_sidebars );
+		}
+
+		// Sidebar ID wp_inactive_widgets is reserved by WP core for inactive widgets.
+		if ( isset( $args['wp_inactive_widgets'] ) ) {
+			unset( $args['wp_inactive_widgets'] );
+		}
+
+		// Check if no registered sidebar.
+		if ( empty( $args ) ) {
```
Comment (danielbachhuber) [inline php/commands/widget.php]: How would you get this error, if we're already checking for `empty( $args )` above?

Comment (ernilambar) [inline php/commands/widget.php]: When `wp widget reset --all` and there is no any registered sidebar in the theme.

**Comment by ernilambar** [comment]
@danielbachhuber build error seems due to tochange in comment in trunk. Should I restart the build?

**Comment by danielbachhuber** [comment]
> build error seems due to tochange in comment in trunk. Should I restart the build?

Merge master to include #3076 and it will be fixed

**Comment by ernilambar** [comment]
Finally `All checks have passed`. :smile:

**Pull request merged**